rasa/nlu/featurizers/sparse_featurizer/lexical_syntactic_featurizer.py

```diff
@@ -119,7 +119,8 @@ def _create_feature_to_idx_dict(
         all_features = []
         for example in training_data.training_examples:
             tokens = example.get(TOKENS_NAMES[TEXT])
-            all_features.append(self._tokens_to_features(tokens))
+            if tokens:
```

Comment:
why do we need this? we didn't have it before

Reply:
because now we can have empty text at the featurization level (e.g., `Message("", {action_name: "action_listen"})`)

Reply:
added a comment about it